script/add-grammar

```diff
@@ -1,124 +1,137 @@
-#!/usr/bin/env ruby
-
-require "optparse"
-require "open3"
-
-ROOT = File.expand_path("../../", __FILE__)
-
-
-# Break a repository URL into its separate components
-def parse_url(input)
-  hosts = "github\.com|bitbucket\.org|gitlab\.com"
-
-  # HTTPS/HTTP link pointing to recognised hosts
-  if input =~ /^(?:https?:\/\/)?(?:[^.@]+@)?(?:www\.)?(#{hosts})\/([^\/]+)\/([^\/]+)/i
-    { host: $1.downcase(), user: $2, repo: $3.sub(/\.git$/, "") }
-  # SSH
-  elsif input =~ /^git@(#{hosts}):([^\/]+)\/([^\/]+)\.git$/i
-    { host: $1.downcase(), user: $2, repo: $3 }
-  # provider:user/repo
-  elsif input =~ /^(github|bitbucket|gitlab):\/?([^\/]+)\/([^\/]+)\/?$/i
-    { host: $1.downcase(), user: $2, repo: $3 }
-  # user/repo - Common GitHub shorthand
-  elsif input =~ /^\/?([^\/]+)\/([^\/]+)\/?$/
-    { host: "github.com", user: $1, repo: $2 }
-  else
-    raise "Unsupported URL: #{input}"
-  end
-end
-
-# Isolate the vendor-name component of a submodule path
-def parse_submodule(name)
-  name =~ /^(?:.*(?:vendor\/)?grammars\/)?([^\/]+)/i
-  path = "vendor/grammars/#{$1}"
-  unless File.exist?("#{ROOT}/" + path)
-    warn "Submodule '#{path}' does not exist. Aborting."
-    exit 1
-  end
-  path
-end
-
-# Print debugging feedback to STDOUT if not running with --quiet
-def log(msg)
-  puts msg if $verbose
-end
-
-def command(*args, hide_warnings: false)
-  log "$ #{args.join(' ')}"
-  stdout, stderr, status = Open3.capture3(*args)
-  unless status.success?
-    output = stdout.strip + "\n" + stderr.strip
-    output.each_line do |line|
-      log "  > #{line}"
-    end
-    warn "Command failed. Aborting."
-    exit 1
-  end
-  return if stderr.empty?
-
-  unless hide_warnings
-    stderr.each_line do |line|
-      log "  > #{line}"
-    end
-  end
-end
-
-usage = """Usage:
-  #{$0} [-q|--quiet] [--replace grammar] url
-Examples:
-  #{$0} https://github.com/Alhadis/language-roff
-  #{$0} --replace sublime-apl https://github.com/Alhadis/language-apl
-"""
-
-$replace = nil
-$verbose = true
-
-OptionParser.new do |opts|
-  opts.banner = usage
-  opts.on("-q", "--quiet", "Do not print output unless there's a failure") do
-    $verbose = false
-  end
-  opts.on("-rSUBMODULE", "--replace=SUBMODDULE", "Replace an existing grammar submodule.") do |name|
-    $replace = name
-  end
-end.parse!
-
-
-$url = ARGV[0]
-
-# No URL? Print a usage message and bail.
-unless $url
-  warn usage
-  exit 1;
-end
-
-# Exit early if docker isn't installed or running.
-log "Checking docker is installed and running"
-command('docker', 'ps')
+#!/bin/sh
```

Comment:
> The use of the `-ef` test on line 89 will clash with this on some systems as `-ef` is not POSIX compliant

I'm aware of that, but on systems whose `test(1)` doesn't implement the `-ef` operator, the worst that can happen is `cd`ing to the current working directory and producing misleading feedback:

~~~sh
if … ! [ . -ef "$root" ] >/dev/null 2>&1; then
	log "Switching directory to $root"
	cd "$root"
fi
~~~

That's why I suppressed output using `>/dev/null 2>&1` and why I tested using `! [` instead of `[ !`. Since it's not doing anything mission critical (it's litle more than UX garnish), I figure it wasn't worth including an explanatory comment for.